core/src/gl/texture.cpp

```diff
@@ -8,226 +8,154 @@
 #include "platform.h"
 #include "util/geom.h"
 
-// Enable only JPEG, PNG, GIF, TGA and PSD
-#define STBI_NO_BMP
-#define STBI_NO_HDR
-#define STBI_NO_PIC
-#define STBI_NO_PNM
-
-#define STB_IMAGE_IMPLEMENTATION
 #include "stb_image.h"
 
 #include <cassert>
 #include <cstring> // for memset
 
 namespace Tangram {
 
-constexpr GLenum TEXTURE_TARGET = GL_TEXTURE_2D;
-
-Texture::Texture(TextureOptions _options) : m_options(_options) {
-}
+Texture::Texture(TextureOptions _options) : m_options(_options) {}
 
-Texture::Texture(const uint8_t* data, size_t length, TextureOptions options) : Texture(options) {
+Texture::Texture(const uint8_t* data, size_t length, TextureOptions options)
+    : Texture(options) {
     loadImageFromMemory(data, length);
 }
 
 Texture::~Texture() {
+    freeBufferData();
+
     if (m_rs) {
         m_rs->queueTextureDeletion(m_glHandle);
     }
 }
 
 bool Texture::loadImageFromMemory(const uint8_t* data, size_t length) {
-    unsigned char* pixels = nullptr;
-    int width = 0, height = 0, comp = 0;
-    pixels = stbi_load_from_memory(data, static_cast<int>(length), &width, &height, &comp, STBI_rgb_alpha);
-
-    if (pixels) {
-        // stbi_load_from_memory loads the image as a series of scanlines starting from
-        // the top-left corner of the image. This call flips the output such that the data
-        // begins at the bottom-left corner, as required for our OpenGL texture coordinates.
-        auto* rgbaPixels = reinterpret_cast<GLuint*>(pixels);
-
-        Texture::flipImageData(rgbaPixels, width, height);
-
-        resize(width, height);
-
-        setPixelData(rgbaPixels, width * height);
-
-        stbi_image_free(pixels);
-
-        return true;
+    freeBufferData();
+
+    // stbi_load_from_memory loads the image as a series of scanlines starting
+    // from the top-left corner of the image. This flips the output such that
+    // the data begins at the bottom-left corner, as required for our OpenGL
+    // texture coordinates.
+    stbi_set_flip_vertically_on_load(true);
+
+    int width = 0, height = 0;
+    int channelsInFile = 0;
+    int channelsRequested = bpp();
+
+    m_buffer = stbi_load_from_memory(data, static_cast<int>(length),
```

Comment:
Does `m_bufferSize` ever get set to match this data?

Reply:
Oh forgot to push the fixup I had..

Reply:
Fixed